utils/hare-fetch-fids

```diff
@@ -0,0 +1,164 @@
+            continue
+        if len(key['key'].split('/')) == 3:
+            nodes.append(json.loads(key['value'])['name'])
+    return nodes
+
+
+def get_keys_for_processes(data: List) -> List[str]:
+    # Get 'm0conf/nodes/<node_id>/processes/<process_fidk>/services/<svc_type>' entries
+    # from the file.  See 'Consul KV Schema' in [4/KV](rfc/4/README.md).
+    regex = re.compile(
+        f'^m0conf\\/.*\\/processes\\/.*\\/services\\/(.+)$')
+    keys = []
+    for key in data:
+        match_result = re.match(regex, key['key'])
+        if not match_result:
+            continue
+        keys.append(match_result.group(0))
+    return keys
+
+
+def add_service_to_node(nodes: List[Node], node_name: str, svc_name: str, fid: str):
+    for node in nodes:
+        if node.name == node_name:
+            node.add_service(svc_name, fid)
+
+
+def get_service_for_node(nodes: List[Node], node_name: str, service: str) -> str:
+    for node in nodes:
+        if node.name == node_name:
+            return node.get_service(service)
+
+
+def process_all_keys(keys: List, nodes: List[Node]):
+    names = {
+        'confd': 'confd',
+        'ios': 'ioservice',
+        'm0_client_s3': 's3server'
+    }
+    for key in keys:
+        # key looks like
+        # 'm0conf/nodes/<name>/processes/<process_fidk>/services/<svc_type>'
+        parts = key.split('/')
+        svc_type = parts[-1]
+        if svc_type in names:
+            node_name = parts[2]
+            process_fidk = parts[-3]
+            add_service_to_node(nodes, node_name, names[svc_type],
+                                to_fid(0x7200000000000001, int(process_fidk)))
+
+
+def read_file():
+    path = '/var/lib/hare/consul-kv.json'
+    with open(path) as consul_kv_file:
+        data = json.load(consul_kv_file)
+    return data
+
+
+def get_nodes_for_cluster() -> List[Node]:
+    file_data = read_file()
+    nodes = []
+    for n in node_names(file_data):
+        nodes.append(Node(n))
+
+    keys = get_keys_for_processes(file_data)
+    process_all_keys(keys, nodes)
+    return nodes
+
+
+def parse_opts(argv):
+    p = argparse.ArgumentParser(description='Fetches the fids for services configured on a node.',
+                                usage='%(prog)s [OPTION]')
+
+    p.add_argument('--service', '-s',
+                   help='service name.'
+                   'List of services- "confd", "ioservice", "s3server".',
+                   required=True,
+                   type=str,
+                   action='store')
+    p.add_argument('--node', '-n',
+                   help='node name',
+                   required=True,
+                   type=str,
+                   action='store')
+
+    return p.parse_args(argv)
+
+
+def main(argv=None):
+    opts = parse_opts(argv)
+    nodes = get_nodes_for_cluster()
+
+    fid = get_service_for_node(nodes, opts.node, opts.service)
+    if fid:
+        print(fid)
+    else:
+        print('Service not found', file=sys.stderr)
```

Comment:
What if service name is right but node name is wrong. It would be good to log an appropriate error.

Reply:
Yes, added logs at appropriate places.

if node_name not found in the node list, then logging -> node-name not found in the cluster.
if service not found in the service list of that node, logging -> service not found.

Reply:
Done, logging error msgs.. where wrong input is given. else raising exceptions